bin/run_docker.py

```diff
@@ -211,51 +172,37 @@ def run_docker(project: str, submission_id: str, log_filename: str = "docker.log
 
         log_text = container
 
-        # Assuming the predictions file is generated in the /output directory
-        predictions_csv_path = os.path.join(os.getcwd(), "output", "predictions.csv")
-        predictions_zip_path = os.path.join(os.getcwd(), "output", "predictions.zip")
-
-        # Retrieve the output CSV or ZIP predictions file and store it on Synapse
-        if os.path.exists(predictions_csv_path):
-            store_predictions_file(
-                syn=syn,
-                project=project,
-                predictions_filename=predictions_csv_path,
-                submission_id=submission_id,
-            )
-        elif os.path.exists(predictions_zip_path):
-            store_predictions_file(
-                syn=syn,
-                project=project,
-                predictions_filename=predictions_zip_path,
-                submission_id=submission_id,
-            )
-
     except Exception as e:
         log_text = str(e).replace("\\n", "\n")
-        create_log_file(log_filename=log_filename, log_text=log_text)
-        store_log_file(
-            syn=syn,
-            project=project,
-            log_filename=f"docker.log",
-            submission_id=submission_id,
+        create_log_file(
+            log_file_name=log_file_name, log_file_path=log_file_path, log_text=log_text
         )
+
         raise
 
     # Create log file and store the log message (``log_text``) inside
-    create_log_file(log_filename=log_filename, log_text=log_text)
-
-    # Store the log file on Synapse under its corresponding folder
-    store_log_file(
-        syn=syn,
-        project=project,
-        log_filename=f"docker.log",
-        submission_id=submission_id,
+    create_log_file(
+        log_file_name=log_file_name, log_file_path=log_file_path, log_text=log_text
     )
 
+    # Rename the predictions file if requested
+    if rename_output:
+        # Get the output directory based on the mounted volumes dictionary used to run the container
+        output_dir = next((key for key in volumes.keys() if "output" in volumes[key]["bind"]), None)
+        if not os.path.exists(output_dir):
+            raise ValueError("Output directory found in ``volumes`` does not exist.")
+
+        # Get the predictions file, assuming the file exists. Otherwise, ``predictions_file = None``
+        predictions_file = next((file for file in glob(os.path.join(output_dir, "predictions.*")) if os.path.exists(file)), None)
+        if predictions_file is None:
+            raise ValueError("No predictions file found in output directory.")
```

Comment:
Continuing [this conversation](https://github.com/Sage-Bionetworks-Workflows/nf-synapse-challenge/pull/18#discussion_r1532399525):

The placeholder `validate.py` does have a check for whether or not there is a predictions file, but yeah it's currently moot since the predictions file path is one of the output channels of `[run_docker.nf](http://run_docker.nf/)`. I can add some logic in `run_docker.py` to capture whether or not there is a predictions output file in the docker execution logs. @BWMac

Reply:
[Jira ticket](https://sagebionetworks.jira.com/browse/IBCDPE-855)